tests/scenario/test_deploy.py

```diff
@@ -47,15 +47,30 @@ def patch_urllib_request():
         "metrics-generator",
     ],
 )
+@patch.object(JujuTopology, "from_charm")
```

Comment:
this is nice! I usually use patch as a context manager but it's often quite hard to remember how to use references.

However, I wonder if for consistency with the existing `ClusterRequirer.get_worker_config` patch, we shouldn't instead do:

```
@patch.object(JujuTopology, "from_charm", MagicMock(return_value=JujuTopology(
        model="test",
        model_uuid="00000000-0000-4000-8000-000000000000",
        application="worker",
        unit="worker/0",
        charm_name="tempo",
    )))
```